tests/lang.sh

```diff
@@ -10,32 +10,49 @@ nix-instantiate --show-trace --eval -E 'builtins.addErrorContext "Hello" (throw
 set +x
 
 fail=0
+just_failed=0
+
+safe_expect () {
```

Comment:
Maybe reuse (and extend?) the existing `expect` function in `tests/common.sh.in` since it does morally the same thing?

It’s marginally less user-friendly as-it-is because it doesn’t allow printing a different error message whether the command crashed or just returned an unexpected result, but I don’t think that’s really a problem

Reply:
Well, It was not possible at first, but now that I discovered that it always returns 1 or 0, `expect 0` and `expect 1` would work just as well, albeit with less explanations.